src/popup/components/TransactionList.vue

```diff
@@ -50,163 +38,99 @@ import {
       ? activeMultisigAccount.value?.gaAccountId
       : activeAccount.value.address);
 
-    const filteredTransactions = computed(
-      () => {
-        const transactionListLocal = [
-          ...getAccountPendingTransactions.value,
-          ...transactions.value.loaded,
-        ];
-
-        if (props.isMultisig && pendingMultisigTransaction.value?.tx) {
-          transactionListLocal.push(pendingMultisigTransaction.value);
-        }
-
-        return transactionListLocal
-          .filter((tr) => {
-            const innerTx = getInnerTransaction(tr.tx);
-            return !props.token
-              || (
-                props.token !== AETERNITY_CONTRACT_ID
-                  ? innerTx?.contractId === props.token
-                  : (!innerTx.contractId || !isFungibleTokenTx(innerTx))
-              );
-          })
-          .filter((tr) => {
-            const { direction, isDex } = useTransactionTx({
-              store: root.$store,
-              tx: tr.tx,
-              externalAddress: tr.transactionOwner,
-            });
-            switch (displayMode.value.key) {
-              case FILTER_MODE.all:
-                return true;
-              case FILTER_MODE.dex:
-                return isDex.value;
-              case FILTER_MODE.out:
-                return direction.value === TX_DIRECTION.sent && !isDex.value;
-              case FILTER_MODE.in:
-                return direction.value === TX_DIRECTION.received;
-              default:
-                throw new Error(`${root.$t('pages.recentTransactions.unknownMode')} ${displayMode.value.key}`);
-            }
-          })
-          .filter(
-            (tr) => !searchPhrase.value || getTxSymbol.value(tr)
-              .toLocaleLowerCase()
-              .includes(searchPhrase.value.toLocaleLowerCase()),
-          )
-          .sort(defaultTransactionSortingCallback)
-          .slice(0, props.maxLength || Infinity);
-      },
-    );
-
-    const showSearchAndFilters = computed(() => (
-      props.showFilters
-      || displayMode.value.key !== FILTER_MODE.all
-      || searchPhrase.value
-    ));
-
-    function checkLoadMore() {
-      if (isDestroyed.value || !transactions.value.nextPageUrl) return;
-      // TODO - use viewport.ts composable after rewriting component to Vue 3
-      const isDesktop = document.documentElement.clientWidth > MOBILE_WIDTH
-          || process.env.IS_EXTENSION;
-      const { scrollHeight, scrollTop, clientHeight } = (isDesktop
-        ? document.querySelector('.app-inner') : document.documentElement)!;
-      if (props.maxLength && filteredTransactions.value.length >= props.maxLength) return;
-      if (scrollHeight - scrollTop <= clientHeight + 100) {
-        // eslint-disable-next-line no-use-before-define
-        loadMore();
-      }
-    }
-
-    async function loadMore() {
-      if (loading.value) return;
+    async function fetchTransactionData(recent?: boolean) {
```

Comment:
This function fetches a list, not a single transaction, so I suggest to use plural or add `List` to the name.